[docs] Add draft AI tutorial (#46779)
expo/expo: docs/pages/tutorial/build-with-ai/build-the-home-screen.mdx

```text Prompt|wrap
Download the image assets for this app from https://docs.expo.dev/static/images/tutorial/sticker-smash-assets.zip and extract them into the assets/images folder, replacing any files with the same names. Then build out the Home screen: show the background-image.png photo large in the center of the screen with rounded corners, and below it two buttons stacked vertically: "Choose a photo" — a prominent button with a yellow border, a white background, a dark label, and a small picture icon — and a plain "Use this photo" button with white text. Use the expo-image library to display the image.
```

<Collapsible summary="Prefer to download the assets yourself?">

If your agent can't download or extract the file, do it manually: download [sticker-smash-assets.zip](/static/images/tutorial/sticker-smash-assets.zip), unzip it, and copy the images into the **assets/images** folder inside your **StickerSmash** folder, replacing files with the same names. Then ask the agent to build the screen with the prompt above, minus its first sentence.

</Collapsible>

> **What you should see**: a photo of mountains at sunset filling most of the screen, with the two buttons below it:

Reviewer: **[critical]** The verify description does not match the asset or the screenshot below it (`style-guide.md#formatting-and-structure`). The asset pack's **background-image.png** and **initial-layout.png** both show a wooden boardwalk over the ocean with palm trees, not mountains at sunset. The verify loop is this tutorial's core mechanic, and a reader checking their phone against this sentence will think the step failed.

```suggestion
**What you should see**: a photo of a wooden boardwalk over the ocean filling most of the screen, with the two buttons below it:
```

<ContentSpotlight
  alt="Home screen with the photo viewer and two buttons"
  src="/static/images/tutorial/initial-layout.png"

>   className="max-w-[720px]"

Reviewer: ```suggestion
  className="max-w-[300px]"
```

This should be 300px otherwise it breaks:

<img width="2912" height="2382" alt="Image" src="https://github.com/user-attachments/assets/5366c6e7-210d-441d-be9e-0c320de02590" />

/>

</Step>

<Step label="3">

## Pick a photo from your library

Time for the first real feature: choosing your own photo. Paste the following prompt into your agent:

```text Prompt|wrap
Add the expo-image-picker library. When I tap "Choose a photo", open my phone's photo library so I can pick an image, and show the one I pick in place of the placeholder photo. If I cancel without picking anything, keep showing the current photo.
```

> **What you should see**: tapping **Choose a photo** opens your phone's photo library. The first time, your phone asks for permission to access your photos — tap **Allow**. The photo you pick replaces the mountains.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`). This also picks up the same boardwalk fix as the comment above: the placeholder photo is not mountains.

```suggestion
**What you should see**: tapping **Choose a photo** opens your phone's photo library. The first time, your phone asks for permission to access your photos. Tap **Allow**. The photo you pick replaces the placeholder photo.
```

</Step>

> > If a result doesn't match what you expected, tell the agent what you see — the [When something looks wrong](/tutorial/build-with-ai/create-your-first-app/#when-something-looks-wrong) section from the previous chapter has the playbook.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
> If a result doesn't match what you expected, tell the agent what you see. The [When something looks wrong](/tutorial/build-with-ai/create-your-first-app/#when-something-looks-wrong) section from the previous chapter has the playbook.
```

## Summary

<ProgressTracker
  currentChapterIndex={2}
  name="BUILD_WITH_AI"

>   summary="The app now has navigation, a styled home screen, and a working photo picker — and you haven't touched a line of code."

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
  summary="The app now has navigation, a styled home screen, and a working photo picker. You haven't touched a line of code."
```
